### DISCUSSION TITLE
lgtf8fx compatability with Arduino AVR timer interrupts

### DISCUSSION THREAD

**Discussion by Oldmaker49** [dbuezas/lgt8fx#297] [General]
Does lgtfx implement timer interrupts in the same way as AVR/Arduino? Will code using timer interrupts written for AVR/Arduino run on lgtfx devices?

**Comment by dbuezas** [comment]
For the most part yes, but you may need to change the prescaler to account for the faster cpu clock.
If you use a timer library, it should work just fine (they usually make timings relative to F_CPU)
If you are configuring the timers by hand, you may find the web tool linked in the readme of this repo (it supports both atmel and lgt)

**Reply by Oldmaker49** [reply]
Thankyou for your reply. 
Just to clarify if the timer interrupt code is writtten for atmel running at 16Mz then it should work with a LGT8fx set to clock divider 2 (16Mhz) at compile?
Congratulations on the timer web tool!- very useful and has help me understand the process.

**Reply by dbuezas** [reply]
Happy to hear you are enjoying the tool!
I also learned a lot creating it, but interestingly I learned more using it.

> if the timer interrupt code is writtten for atmel running at 16Mz then it should work with a LGT8fx set to clock divider 2 (16Mhz) at compile

YES, that should be the case. The other way around may not hold since LGT timers have many more features.
